x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/policy_json_flyout.tsx

```diff
@@ -18,29 +18,35 @@ import {
   EuiText,
   EuiTitle,
 } from '@elastic/eui';
-import { Policy } from '../../../services/policies/types';
+import { Policy, PolicyFromES } from '../../../services/policies/types';
+import { serializePolicy } from '../../../services/policies/policy_serialization';
 
 interface Props {
   close: () => void;
   policy: Policy;
+  existingPolicy?: PolicyFromES;
   policyName: string;
 }
 
-export const PolicyJsonFlyout: React.FunctionComponent<Props> = ({ close, policy, policyName }) => {
-  const getEsJson = ({ phases }: Policy) => {
-    return JSON.stringify(
-      {
-        policy: {
-          phases,
-        },
+export const PolicyJsonFlyout: React.FunctionComponent<Props> = ({
+  close,
+  policy,
+  policyName,
+  existingPolicy,
+}) => {
+  const { phases } = serializePolicy(policy, existingPolicy?.policy);
```

Comment:
nit; if you name this variable `policy` (without destructuring) you can assign it directly to the object being stringified

Reply:
Thanks for the suggestion, but I think we have a `name` field in `PolicyFromES` which can't be included in the request, that is why we only use `phases` variable.